File: src/main/java/com/skylivings/webapp/model/Property.java
package com.skylivings.webapp.model;

import java.util.Date;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;

import com.skylivings.webapp.model.enums.PropertyStatus;
import com.skylivings.webapp.model.enums.PropertyType;

import lombok.Data;

@Data
@Entity
public class Property {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int propertyId;
	private String propertyCode;
	private String propertyName;
	@Enumerated(EnumType.STRING)
	private PropertyType propertyType;
	@Enumerated(EnumType.STRING)
	private PropertyStatus propertyStatus;
	private Date onBoardingDate;
	@OneToOne(cascade = CascadeType.PERSIST)
	private Address address;
	@ManyToOne(cascade = CascadeType.PERSIST)
	@JoinColumn(name = "managerId")
	private PropertyManager manager;
	@OneToMany
	private List<Room> rooms;
	@ManyToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "ownerId")
	private PropertyOwner owner;
}

File: skyfront/src/app/classes/property.ts
import { Manager } from "./manager";
import { Room } from "./room";

export class Property {
  propertyId!:number;
  propertyCode!: string;
  propertyType!: string;
  propertyName!: string;
  propertyStatus!: string;
  buildingName!: string;
  streetName!: string;
  city !: string;
  state !: string;
  country !: string;
  zip !: number;
  onBoardingDate!: Date;
  rooms!: Room[];
  ownerId!: number;
  firstName!: string;
  lastName!: string;
  contact!: string;
  alternativeContact!: string;
  email!: string;
  gender!: string;
  manager!: Manager;
}
